catalyst_voices/packages/internal/catalyst_voices_services/lib/src/user/user_service.dart

```diff
@@ -1,86 +1,59 @@
 import 'dart:async';
 
 import 'package:catalyst_voices_models/catalyst_voices_models.dart';
-import 'package:catalyst_voices_services/catalyst_voices_services.dart';
+import 'package:catalyst_voices_repositories/catalyst_voices_repositories.dart';
 import 'package:catalyst_voices_shared/catalyst_voices_shared.dart';
 
 abstract interface class UserService implements ActiveAware {
   factory UserService({
-    required KeychainProvider keychainProvider,
-    required UserStorage userStorage,
-    required DummyUserFactory dummyUserFactory,
+    required UserRepository userRepository,
   }) {
     return UserServiceImpl(
-      keychainProvider,
-      userStorage,
-      dummyUserFactory,
+      userRepository,
     );
   }
 
   Account? get account;
 
-  Stream<Account?> get watchAccount;
-
-  Keychain? get keychain;
+  List<Account> get accounts;
 
-  Future<List<Keychain>> get keychains;
+  Stream<Account?> get watchAccount;
 
-  Stream<Keychain?> get watchKeychain;
+  Future<User> getUser();
 
   Future<void> useLastAccount();
 
   Future<void> useAccount(Account account);
 
-  Future<void> useKeychain(String id);
-
-  Future<void> removeCurrentKeychain();
-
-  Future<void> removeKeychain(String id);
+  Future<void> removeAccount(Account account);
 
   Future<void> dispose();
 }
 
 final class UserServiceImpl implements UserService {
-  final KeychainProvider _keychainProvider;
-  final UserStorage _userStorage;
-  final DummyUserFactory _dummyUserFactory;
+  final UserRepository _userRepository;
 
   final _logger = Logger('UserService');
 
-  User? _user;
-  final _userSC = StreamController<User?>.broadcast();
-
-  Keychain? _keychain;
-  final _keychainSC = StreamController<Keychain?>.broadcast();
-  StreamSubscription<bool>? _keychainUnlockSub;
+  User _user = const User(accounts: []);
+  final _userSC = StreamController<User>.broadcast();
 
   bool _isActive = true;
 
   UserServiceImpl(
-    this._keychainProvider,
-    this._userStorage,
-    this._dummyUserFactory,
+    this._userRepository,
   );
 
   @override
-  Account? get account => _user?.activeAccount;
+  Account? get account => _user.activeAccount;
 
   @override
-  Stream<Account?> get watchAccount async* {
-    yield account;
-    yield* _userSC.stream.map((user) => user?.activeAccount).distinct();
-  }
-
-  @override
-  Keychain? get keychain => _keychain;
+  List<Account> get accounts => List.unmodifiable(_user.accounts);
 
   @override
-  Future<List<Keychain>> get keychains => _keychainProvider.getAll();
-
-  @override
-  Stream<Keychain?> get watchKeychain async* {
-    yield _keychain;
-    yield* _keychainSC.stream;
+  Stream<Account?> get watchAccount async* {
+    yield account;
+    yield* _userSC.stream.map((user) => user.activeAccount).distinct();
   }
```

Comment:
I've noticed one problem with this pattern of using `async*/yield*`, it creates a stream that is not a broadcast stream so you can only listen to it once, each time you want to listen again you need to call watchAccount again. In most cases this shouldn't be a problem but if you're passing it somewhere then it might be listened twice without you doing it deliberately. 

The safest imho is using the [BehaviorSubject](https://pub.dev/documentation/rxdart/latest/rx/BehaviorSubject-class.html) from rxdart, it creates a broadcast stream but also replays the last value when you subscribe.

Not requesting any change but just sharing my finding

Reply:
Interesting, didn't know it! Thanks for sharing